test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpMetricsExporterTests.cs

```diff
@@ -0,0 +1,150 @@
+// <copyright file="OtlpMetricsExporterTests.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Metrics;
+using System.Linq;
+using System.Threading;
+using OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Resources;
+using OpenTelemetry.Tests;
+using OpenTelemetry.Trace;
+using Xunit;
+using GrpcCore = Grpc.Core;
+using OtlpCollector = Opentelemetry.Proto.Collector.Metrics.V1;
+using OtlpMetrics = Opentelemetry.Proto.Metrics.V1;
+
+namespace OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests
+{
+    public class OtlpMetricsExporterTests
+    {
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void ToOtlpResourceMetricsTest(bool addResource)
+        {
+            using var exporter = new OtlpMetricsExporter(
+                new OtlpExporterOptions(),
+                new NoopMetricsServiceClient());
+
+            if (addResource)
+            {
+                exporter.SetResource(
+                    ResourceBuilder.CreateEmpty().AddAttributes(
+                        new List<KeyValuePair<string, object>>
+                        {
+                            new KeyValuePair<string, object>(ResourceSemanticConventions.AttributeServiceName, "service-name"),
+                            new KeyValuePair<string, object>(ResourceSemanticConventions.AttributeServiceNamespace, "ns1"),
+                        }).Build());
+            }
+            else
+            {
+                exporter.SetResource(Resource.Empty);
```

Comment:
do we need to set this to Emtpy?

Reply:
Yes, this tests the following logic (copied from the trace exporter - I plan to take a pass and de-dupe some code between the trace and metric exporter).

https://github.com/open-telemetry/opentelemetry-dotnet/blob/7c611c8537d4d2c780dbc5fa999a3b3b7fd0d12d/src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpMetricsExporter.cs#L155-L164

The [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/ccbfdd9020ba0d581aa5d880235302f8b0eb8669/specification/resource/semantic_conventions/README.md#service) requires `service.name` to fall back to `unknown_service:...` when it is not set. I _think_ it's possible for someone to create a resource without adding a `service.name`. Technically, we could probably solve this upstream of the exporter and then we wouldn't need this code ehre.